internal/services/mongocluster/mongo_cluster_resource_test.go

```diff
@@ -257,6 +289,123 @@ resource "azurerm_mongo_cluster" "geo_replica" {
 `, source, data.RandomInteger, data.Locations.Secondary)
 }
 
+func (r MongoClusterResource) pointInTimeRestore(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_mongo_cluster" "point_in_time_restore" {
+  name                   = "acctest-mc-restore%d"
+  resource_group_name    = azurerm_resource_group.test.name
+  location               = azurerm_resource_group.test.location
+  create_mode            = "PointInTimeRestore"
+  administrator_username = "adminTerraform"
+  administrator_password = "QAZwsx123update"
+
+  restore {
+    source_id         = azurerm_mongo_cluster.test.id
+    point_in_time_utc = timeadd(timestamp(), "-1s")
+  }
+
+  // As "point_in_time_utc" is retrieved dynamically, so it would cause diff when tf plan. So we have to ignore it here.
```

Comment:
This can be solved with time_static resource, here is an example https://github.com/hashicorp/terraform-provider-azurerm/blob/acba5de274641085b17197dfb3820de1ed62f82d/internal/services/authorization/pim_active_role_assignment_resource_test.go#L274